hardware/bluetooth/control/bluetooth_control_command.ksy

```diff
@@ -0,0 +1,42 @@
+meta:
+  id: bluetooth_control_command
+  title: Bluetooth Host Controller Interface Control Protocol
+  license: Unlicense
+  file-extension:
+    - hcd
+  endian: le
+  imports:
+    - vendor_specific/bluetooth_control_group_vendor_specific
+    - ../bluetooth_vendors_ids
+doc-ref: https://www.bluetooth.org/docman/handlers/downloaddoc.ashx?doc_id=478726
+params:
+  - id: vendor
+    type: u2
+    enum: bluetooth_vendors_ids::vendor
+seq:
+  - id: opcode
+    type: opcode
+  - id: size
+    type: u1
+  - id: parameters
+    size: size
+    type:
+      switch-on: opcode.group
+      cases:
+        "group::vendor_specific": bluetooth_control_group_vendor_specific(vendor, opcode.command)
+types:
+  opcode:
+    seq:
+      - id: packed
+        type: u2
+    instances:
+      group:
+        -orig-id: OpCode Group Field
+        value: packed >> 10
+        enum: group
+      command:
+        -orig-id: OpCode Command Field
+        value: packed & 0x3FF
```

Comment:
Is there a reason why you don't use the little-endian bit-sized integers (https://github.com/kaitai-io/kaitai_struct/issues/155) instead, when we already have them implemented? Compatibility with stable 0.8 can be hardly a decent motive why use this old-school approach, because you already use lists in `doc-ref`.

I think relying on version 0.9 is OK, many existing specs also do it, and hopefully the 0.9 version will be released soon.

```suggestion
    meta:
      bit-endian: le
    seq:
      - id: command
        -orig-id: OpCode Command Field
        type: b10
      - id: group
        -orig-id: OpCode Group Field
        type: b6
        enum: group
```

Reply:
Is it implemented for all the target languages, the most importantly python and C++?

And, BTW, why not just `endian` or even not my endianness encoding scheme (the links to the impl in python (it has to be ported to scala to be usable, it makes no sense to move these computations to the runtime) are present in the issue)?

Reply:
> Is it implemented for all the target languages, the most importantly python and C++?

Yes, see https://github.com/kaitai-io/kaitai_struct/issues/155#issuecomment-623027660 and specifically kaitai-io/kaitai_struct_python_runtime@c5ff9a1e397aa3307ba2a0e217b58dfe898cbeed and kaitai-io/kaitai_struct_cpp_stl_runtime@37a8d3bc137313b9672fdf367bc942cb8691853b.

> why not just `endian` or even not my endianness encoding scheme

This has been discussed a few months ago in kaitai-io/kaitai_struct#155 and the implementation work has already been done - it's a bit late now for alternative feature suggestions... A separate `bit-endian` attribute is needed because of backwards compatibility - `endian` previously hasn't affected bit fields, so changing that would silently break some existing specs. The two options for `bit-endian` are enough to support all possible bit/byte order combinations (see https://github.com/kaitai-io/kaitai_struct/issues/155#issuecomment-615345304), so a more complex format for specifying endianness shouldn't be necessary.

Reply:
>it's a bit late now for alternative feature suggestions...

OK. IMHO it still makes sense to just break the compatibility and just fix the specs in this repo and just force the authors to fix their specs, rather than to accumulate tech debt and mess (I cannot imagine real world scenarios where numbers are LE but bit fields are used as BE and IMHO the name chosen is very containtuitive).

Thanks for the suggestion and the clarification. Should probably apply it to my other specs too